templates/boltzmann/core/prelude.ts

```diff
@@ -47,12 +53,74 @@ if (!process.env.HONEYCOMB_TEAM && process.env.HONEYCOMBIO_TEAM) {
   process.env.HONEYCOMB_TEAM = process.env.HONEYCOMBIO_TEAM
 }
 
-beeline({
-  writeKey: process.env.HONEYCOMB_WRITEKEY,
-  dataset: process.env.HONEYCOMB_DATASET,
-  sampleRate: Number(process.env.HONEYCOMB_SAMPLE_RATE) || 1,
-  serviceName,
-})
+if (!process.env.HONEYCOMB_DATASET && process.env.HONEYCOMBIO_DATASET) {
+  process.env.HONEYCOMB_DATASET = process.env.HONEYCOMBIO_DATASET
+}
+
+let sdk: NodeSDK | null = null
+
+if (
+  process.env.HONEYCOMB_WRITE_KEY &&
+  process.env.HONEYCOMB_DATASET &&
+  process.env.HONEYCOMB_API_HOST
+) {
+  const metadata = new Metadata()
+  metadata.set('x-honeycomb-team', process.env.HONEYCOMB_WRITE_KEY)
+  metadata.set('x-honeycomb-dataset', process.env.HONEYCOMB_DATASET)
```

Comment:
This bit is a little cargo-culted from what we did in the CI repo. I believe this in particular is necessary boilerplate for setting up honeycomb, but I haven't proven that to myself. I also believe these are sufficient and that I'm not missing any, but definitely check me here!